### PR TITLE
Sort reactions on the timeline by count and then timestamp and show the date on the summary view

### PR DESCRIPTION
**Pull request by langleyd**
# What's in this PR
- Sort reaction aggregations on the timeline by count and then by most recent timestamp ascending(tagging new reactions on the end)
- Show the timestamp on the summary view and sort descending(newest at the top)

# What does it look like

### CODE REVIEW

File: ElementX/Sources/Services/Timeline/TimelineItemContent/AggregratedReaction.swift [hunk 1]
```diff
@@ -23,7 +23,15 @@ struct AggregatedReaction: Hashable {
...
     let key: String
     /// The user ids of those who sent the reactions
-    let senders: [String]
+    let senders: [ReactionSender]
+}
+
+/// Details of who sent the reaction
+struct ReactionSender: Hashable {
+    /// The id of the user who sent the reaction
+    let senderId: String
```
Comment (pixlwave) [inline ElementX/Sources/Services/Timeline/TimelineItemContent/AggregratedReaction.swift:32]: We generally use `ID` on the Swift side :)
```suggestion
    let senderID: String
```

File: ElementX/Sources/Services/Timeline/TimelineItems/RoomTimelineItemFactory.swift [hunk 2]
```diff
@@ -313,13 +313,22 @@ struct RoomTimelineItemFactory: RoomTimelineItemFactoryProtocol {
...
+            return AggregatedReaction(accountOwnerID: userID, key: reaction.key, senders: senders)
         }
         .sorted { a, b in
-            // Sort by count and then by key for a consistence experience.
-            // Otherwise emojis can switch around. We can replace
-            // with timestamp as a secondary sort when it is available.
-            (a.count, a.key) > (b.count, b.key)
+            // Sort aggregated reactions by count and then by most recent reaction
+            if a.count == b.count {
+                return a.senders[0].timestamp < b.senders[0].timestamp
```
Comment (pixlwave) [inline ElementX/Sources/Services/Timeline/TimelineItems/RoomTimelineItemFactory.swift:329]: Should this be the other way around for most recent?
```suggestion
                return a.senders[0].timestamp > b.senders[0].timestamp
```

**Review by pixlwave — APPROVED** [review]
LGTM 😎

File: ElementX/Sources/Services/Timeline/TimelineItems/RoomTimelineItemFactory.swift (hunk 2 above)
Comment (langleyd) [reply ElementX/Sources/Services/Timeline/TimelineItems/RoomTimelineItemFactory.swift:329]: The comment should be clearer. It's sorted by count and then by aggregation with most recent reaction(index zero in the aggregation's list of reactions, which are in descending order from line 321) in ascending order(so new reactions are added to the end of the reaction layout).

I'll write that in a clearer way :)

Comment (pixlwave) [reply ElementX/Sources/Services/Timeline/TimelineItems/RoomTimelineItemFactory.swift:329]: Right, understood 👍

**Comment by aringenbach** [comment]
Fixes #1289

**Pull request merged**